README.md

```diff
@@ -1 +1,51 @@
 # Modbus Processing Unit (MPU)
+
+Provides a common modbus communication library, which can be left running in
+FPGA and provide telemetry. The major idea is to create in FPGA a simple CPU (a
+Turing Machine if you want), narrowly designed into reading out data from
+Modbus devices. The design is:
+
+* input FIFO with commands
+* array storage for commands (instructions stacks), which can be updated from
+  FIFO when one loop of the library finished
+* processing unit. That takes commands from instructions stacks, and executed
+  them
+* output array. There responses from Modbus devices are stored, as commanded
+  (readout command)
+* output to telemetry FIFO, with commands to update telemetry memory with
+  values readout from device
+* optional raw output to output FIFO
+
+Those instructions are supported:
+
+0. stop executing and wait for a new instructions in FIFO
+1. write the next byte to the port
+3. wait for given ticks/ms/us/..
+4. read from the port and store data into output array
+4. loop (jump to some instruction, usually 0 to repeat the sequence; with
+   looping, a check for commanding FIFO will be provided, and if that contains
+   new instructions, instruction stack will be replaced with those new
+   instructions)
+5. write data to output FIFO
+x. check CRC of data stored in output array
```

Comment:
What is the CRC? Please consider to put the full name for abbreviation if it appears as a first time in the document?

Reply:
Added link to Wikipedia article.